Updates to the release checklists (#29227)
mantidproject/mantid: dev-docs/source/ReleaseChecklist.rst

Timeline
########

Releases are normally planned to occur on a Monday, therefore this
page will be refer to days assuming that is correct, if the final
release day is not planned to be Monday then the names of the days
will have to be changed.

.. note::
   We used to release on a Friday, but changed to a Monday as no support was available over the weekend in case issues
   were raised with the newly released version.

Friday, Release-4 weeks (one week before code freeze)
#########################################################

**Task Priorities**: Development, Testing, Documentation.

*  Post on the General slack channel reminding developers of the
   impending release and stating that they have only 5 days left before
   the code freeze.
*  Send an email to beta test users explaining the dates for the
   testing, and stating they will have more detail on the start of the
   first day (cc the Local PMs so they can organise a similar message at their facilities).

> *  Before the code freeze is in place the PM and TSC need to agree the maintenace tasks for the next release period.

Reviewer: ```suggestion
*  Before the code freeze is in place the PM and TSC need to agree the maintenance tasks for the next release period.
```

Code Freeze
-----------

*  Post on the General slack channel asking everyone to ensure they

>    have move any incomplete issues, stating the code freeze is in place, and

Reviewer: ```suggestion
   have moved any incomplete issues to the next milestone, stating the code freeze is in place, and
```

>    warning developers that non-blocker tickets will be moved from the

Reviewer: ```suggestion
   warning developers that non-blocker issues will be moved from the
```

   milestone on Monday morning.
*  Final Testing afternoon, attempt to drive the pull requests for this
   milestone down to 0.

Monday, Release-2 weeks & 4 days
################################

**Task Priorities**: Blocker bug fixes, Testing, Release Notes.

Clearing the project board
--------------------------

* Go through the Zenhub project board for the release milestone (not the sprint milestone), ensuring that:

 *  All issues are intended for the release.
 *  Any new issues are triaged on a daily basis, and allocated to staff.
 *  Issues that are not important for the release should be moved to a more appropriate milestone.

>     Don't leave anything in the release milestone that is not definitely for that release.

Reviewer: :+1:

* Send an email, including the text of the release notes, to the
  following lists
* ``nobugs@nobugsconference.org``
* ``news@neutronsources.org``
* ``neutron@neutronsources.org``

> * Also post the contents of the message on General on Slack

Reviewer: I think we have been using Annoucements for this too.

Author: Good point I've changed this

* Create a new item on the forum news

> * Close the release milestone in the github

Reviewer: ```suggestion
* Close the release milestone on github
```
